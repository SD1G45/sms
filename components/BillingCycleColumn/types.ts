export interface BillingCycleColumnProps extends React.HTMLAttributes<HTMLInputElement> {
  spentAmount: string;
  salesAmount: string;
  billingCycle: string;
  billingCycleRoute: string;
}