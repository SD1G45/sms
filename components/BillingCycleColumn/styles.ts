import styled from "styled-components";
import { ColumnDiv } from "../../page-styles/dashboard/styles";

export const SalesColumnDiv = styled(ColumnDiv)`
  border-left: 1px solid #BEC8D2;
  justify-content: center;
  padding: 0px;
`;

export const SalesDivider = styled.div`
  width: 100%; 
  border-bottom: 1px solid #BEC8D2;
  margin: 15px 0px 7px 0px;
`;