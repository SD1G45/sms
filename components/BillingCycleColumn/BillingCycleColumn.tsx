import React from "react";
import BillingCycleInfo from "../BillingCycleInfo";
import { SalesColumnDiv, SalesDivider } from "./styles";
import { BillingCycleColumnProps } from "./types";

const BillingCycleColumn: React.FC<BillingCycleColumnProps> = ({
  salesAmount,
  spentAmount,
  billingCycle,
  billingCycleRoute
}) => {
  return (
    <SalesColumnDiv>
      <BillingCycleInfo
        title="Amount Spent"
        viewLinkText="View billing cycle"
        amount={spentAmount}
        subtitle={"Billing cycle: " + billingCycle}
        linkRoute={billingCycleRoute}
      />
      <SalesDivider/>
      <BillingCycleInfo
        title="Estimated Sales"
        amount={salesAmount}
        subtitle="*Estimation only*"
      />
    </SalesColumnDiv>
  );
};

export default BillingCycleColumn;