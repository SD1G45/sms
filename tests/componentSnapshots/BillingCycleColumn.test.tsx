import React from 'react';
import renderer from "react-test-renderer";
import "jest-styled-components";
import BillingCycleColumn from "../../components/BillingCycleColumn";

test('BillingCycleColumn renders correctly', () => {
  const tree = renderer
    .create(
      <BillingCycleColumn
        spentAmount="$1,180.00"
        salesAmount="$175.00"
        billingCycle="10/07/21 - 10/31/21"
        billingCycleRoute="/billing"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});