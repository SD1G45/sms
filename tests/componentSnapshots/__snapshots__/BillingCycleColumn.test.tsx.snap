// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingCycleColumn renders correctly 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <div
      className=""
    >
      <div
        className=""
      >
        <text
          className=""
        >
          Amount Spent
        </text>
        <text
          className=""
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          View billing cycle
        </text>
      </div>
      <text
        className=""
      >
        $1,180.00
      </text>
      <text
        className=""
      >
        Billing cycle: 10/07/21 - 10/31/21
      </text>
    </div>
  </div>
  <div
    className=""
  />
  <div
    className=""
  >
    <div
      className=""
    >
      <div
        className=""
      >
        <text
          className=""
        >
          Estimated Sales
        </text>
      </div>
      <text
        className=""
      >
        $175.00
      </text>
      <text
        className=""
      >
        *Estimation only*
      </text>
    </div>
  </div>
</div>
`;
